internal/repository/inputreader.go

```diff
@@ -0,0 +1,126 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package repository
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/cartesi/rollups-node/internal/node/model"
+	"github.com/jackc/pgx/v5"
+)
+
+func (pg *database) InsertInputsAndUpdateLastProcessedBlock(
+	ctx context.Context,
+	inputs []Input,
+	blockNumber uint64,
+	appAddress Address,
+) error {
+	query := `
+	INSERT INTO input
+		(index,
+		status,
+		blob,
+		block_number,
+		application_address)
+	VALUES
+		(@index,
+		@status,
+		@blob,
+		@blockNumber,
+		@appAddress)`
+
+	query2 := `
+	UPDATE application
+	SET last_processed_block = @blockNumber
+	WHERE
+		application_address=@appAddress`
+
+	args := pgx.NamedArgs{
+		"blockNumber": blockNumber,
+		"appAddress":  appAddress,
+	}
+
+	tx, err := pg.db.Begin(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to insert inputs: %w\n", err)
+	}
+
+	for _, input := range inputs {
+		inputArgs := pgx.NamedArgs{
+			"index":       input.Index,
+			"status":      input.CompletionStatus,
+			"blob":        input.Blob,
+			"blockNumber": input.BlockNumber,
+			"appAddress":  input.AppAddress,
+		}
+		_, err = tx.Exec(ctx, query, inputArgs)
+		if err != nil {
+			return fmt.Errorf("unable to insert inputs: %w\n", err)
+		}
+	}
+
+	_, err = tx.Exec(ctx, query2, args)
+	if err != nil {
+		return fmt.Errorf("unable to insert inputs: %w\n", err)
+	}
+
+	err = tx.Commit(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to insert inputs: %w\n", err)
+	}
+
+	return nil
+}
+
+func (pg *database) GetAllApplications(
```

Comment:
This should be changed or another method added , so it returns only the "running" Applications